### PR TITLE
Drop support for PHP 5.3

### LINKED ISSUE #406: Deprecate `Stripe\JsonSerializable` and bump minimum version to PHP 5.4 (or later)
PHP version 5.3 is at End of Life cycle for at least 3 years now. The file `./lib/JsonSerializable.php` as a bridge for the `\JsonSerializable` interface prevents execution of the AspectMock library for testing, apart from not being PSR-2 compliant.

[AspectMock](https://github.com/Codeception/AspectMock) library from Codeception is a powerful mocking library we use to test static methods and mock objects but it parses PHP files on their own and doesn't recognise the special syntax used (`interface` declarations must be at root).

This issue is a suggestion to make the current version usable much like the v1.18.0 version for PHP 5.2, as specified in [Legacy Version Support](https://github.com/stripe/stripe-php/#legacy-version-support).

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @brandur-stripe 
cc @stripe/api-libraries 

This PR drops support for PHP 5.3, and gets rid of some now unneeded stuff.

Note that the destination branch is `integration-v6` instead of `master` so that we can merge all PRs going into stripe-php 6.0.0 and solve conflicts before merging into `master`.

Fixes #406, although the fix won't be effective until we merge into `master` and actually release a new major version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,12 @@ Please see https://stripe.com/docs/api for up-to-date documentation.
 
 ## Legacy Version Support
 
+### PHP 5.3
+
+If you are using PHP 5.3, you can download v5.8.0 ([zip](https://github.com/stripe/stripe-php/archive/v5.8.0.zip), [tar.gz](https://github.com/stripe/stripe-php/archive/v5.8.0.tar.gz)) from our [releases page](https://github.com/stripe/stripe-php/releases). This version will continue to work with new versions of the Stripe API for all common uses.
```
Comment (remi-stripe) [inline README.md]: Is it worth adding a scary blurb about how bad using this version (and 5.2) is?

File: .travis.yml [hunk 2]
```diff
@@ -2,52 +2,27 @@ sudo: false
...
+  - 7.1
+  - 7.2
+  - hhvm
+
+env:
+  global:
+    - STRIPE_MOCK_VERSION=0.5.0
+  matrix:
+    - AUTOLOAD=1
+    - AUTOLOAD=0
```
Comment (brandur-stripe) [inline .travis.yml]: Thanks for changing this whole thing back :)

File: README.md (hunk 1 above)
Comment (brandur-stripe) [reply README.md]: Couldn't hurt!

**Comment by brandur-stripe** [comment]
LGTM.

**Comment by arcanedev-maroc** [comment]
How about the **short array syntax** in PHP 5.4 ?

**Comment by ob-stripe** [comment]
@arcanedev-maroc Yep, that's on my TODO list too! :)

**Pull request merged**